Mathlib/RingTheory/HopkinsLevitzki.lean

```diff
@@ -29,32 +29,50 @@ import Mathlib.RingTheory.Spectrum.Prime.Noetherian
 * [F. Lorenz, *Algebra: Volume II: Fields with Structure, Algebras and Advanced Topics*][Lorenz2008]
 -/
 
-variable {R M : Type*} [Ring R] [AddCommGroup M] [Module R M]
+variable {R₀ R M : Type*} [Ring R₀] [Ring R] [Module R₀ R]
+  [AddCommGroup M] [Module R₀ M] [Module R M] [IsScalarTower R₀ R R] [IsScalarTower R₀ R M]
 
 open Ideal in
-theorem IsSemiprimaryRing.isNoetherian_iff_isArtinian [IsSemiprimaryRing R] :
-    IsNoetherian R M ↔ IsArtinian R M := by
+theorem IsSemiprimaryRing.isArtinian_imp_finite_and_isNoetherian_iff_isArtinian
+    [IsSemiprimaryRing R] [Module.Finite R₀ (R ⧸ Ring.jacobson R)] :
+    (IsArtinian R M → Module.Finite R₀ M) ∧ (IsNoetherian R M ↔ IsArtinian R M) := by
```

Comment:
Do you think it is worth stating the first half as a separate lemma?

Reply:
Seems reasonable to me.

Reply:
Maybe I'll extract an induction principle out of this and deduce separate lemmas from it ... I think I can show a semiprimary ring is Noetherian (equivalently, Artinian) iff the Jacobson radical is fg. I'll make a separate PR from branch#IsTwoSided_finite for interaction of finiteness (and more) with IsTwoSided.

Reply:
The induction principle has been extracted; the `IsTwoSided_finite` branch is PR'd separately as #22151.